src/transformers/trainer.py

```diff
@@ -1725,6 +1725,9 @@ def _hp_search_setup(self, trial: Union["optuna.Trial", Dict[str, Any]]):
         if self.is_deepspeed_enabled:
             if self.args.deepspeed is None:
                 raise ValueError("For sweeps with deepspeed, `args.deepspeed` must be set")
+
+            self.accelerator.free_memory()
+
```

Comment:
why have you added that here ?

Reply:
I had an issue where the memory allocated by the Deepspeed engine was not properly released. I found that calling `engine.destroy()` fixed the problem partially (`accelerator.free_memory()` calls `engine.destroy()`).

At the moment, we still have memory leakage but I believe the issue is on Deepspeed side. I added some `del` statements in the `destroy` function of the deepspeed engine which solved the issue and I will try to propose a PR there.

I put the call to `free_memory` here because the engine is reset just after that (in `self.create_accelerator_and_postprocess()`).